### DISCUSSION TITLE
how to install streamlit in termux?

### DISCUSSION THREAD

**Discussion by leechaowen** [termux/termux-packages#8036] [Q&A] [answered]
i know there is a way to install streamlit in termux, First ,install ubuntu by proot,then install streamlit in ubuntu, but it's too low speed.
Any idea install streamlit in termux by native ? such as by run command pkg install streamlit . Thanks your attention.

**Comment by Wadjinny** [comment, marked as answer]
I faced the same problem, and after a long journey on stack-overflow, i found it. So basically you'll have to run ` pkg install libzmq ` and then install streamlit:
` pip install streamlit ` and hopefully, no errors comes up.

**Reply by Danial-ilyasa** [reply]
how come I can't what's wrong

**Comment by hreikin** [comment]
I installed streamlit today and encountered some errors along the way, I am unable to edit the wiki with my solutions as I don't have the correct permissions so figured i would document it here instead. Hopefully someone will update the wiki with the relevant information.

When using `pip install streamlit` the process will fail at a couple of points, first when building the wheel for numpy and then when building the wheel for pyarrow. To get `streamlit` to install successfully i had to do the following:

```bash
pkg install libarrow-cpp    # Fix for pyarrow
export MATHLIB="m"          # Fix for numpy
pip install streamlit       # Install streamlit
```

The next error i encountered came when trying to run the demo with `streamlit hello`, pyarrow complained about missing symbols; `dlopen failed: cannot locate symbol "PyBaseObject_Type"`. To fix this the following is required so pyarrow can find the required symbols:

```bash
export LDFLAGS="-lpython3.11"    # Fix for pyarrow missing symbols
streamlit hello                  # Run streamlit
```
